File: frontend/src/entities/waste.ts
import { EntityId } from "./common";

export type Waste = "Vegetale" | "Secco" | "Umido" | "VPL" | "Carta";

export type Municipality = {
  id: EntityId;
  name: string;
  zone: string;
  area: string;
  created_at: Date;
  updated_at: Date;
};

export type WasteCollection = {
  id: EntityId;
  date: Date;
  waste: Waste;
  created_at: Date;
  updated_at: Date;
  municipality_id: EntityId;
};

File: scraper/scraper/entities/waste.py
from typing import Literal, List

Waste = Literal["Vegetale", "Secco", "Umido", "VPL", "Carta"]


class CollectionSchedule:
    def __init__(self, date: str, wastes: List[Waste]) -> None:
        self._date = date
        self._wastes = wastes

    @property
    def date(self) -> str:
        return self._date

    def __str__(self):
        return f"{self.date}: {self._wastes}"
